lib/Sema/TypeChecker.h

```diff
@@ -487,6 +487,13 @@ void diagnoseDuplicateCaptureVars(CaptureListExpr *expr);
 Type checkReferenceOwnershipAttr(VarDecl *D, Type interfaceType,
                                  ReferenceOwnershipAttr *attr);
 
+/// Checks if the given witness can structurally match the requirement, ie.
+/// whether it has a matching kind like static vs. non-static, matching decl,
+/// etc. It does not check if the types of the witness is able to satisfy the
+/// requirement.
+/// If there is a matching failure, returns a `true`, otherwise return `false`
+bool witnessStructureMatches(ValueDecl *req, const ValueDecl *witness);
```

Comment:
This should be the opposite, right? i.e returns `true` on success

Reply:
Indeed. Thanks for catching.